src/network/url_downloader.cpp

```diff
@@ -99,21 +99,57 @@ QByteArray download(QNetworkAccessManager* manager, const Time& timeout, QUrl co
 
         const auto msg = reply->errorString().toStdString();
 
+        if (reply->error() == QNetworkReply::ProxyAuthenticationRequiredError)
+            reply->abort();
+
         if (abort_download)
             throw mp::AbortedDownloadException{msg};
         else
             throw mp::DownloadException{url.toString().toStdString(), download_timeout.isActive() ? msg : "Network timeout"};
     }
     return reply->readAll();
 }
+
+std::unique_ptr<QNetworkProxy> discover_http_proxy()
```

Comment:
Wouldn't this unit deserve tests?

Reply:
I gave this quite a bit of thought and we would basically have to get into the business of mocking a proxy somehow.  And that I don't think is worth the effort for something like this.

Reply:
I was thinking something much more superficial, like just checking whether or not the application had a `QNetworkProxy` proxy registered with the right url and data, depending on whether/how `http_proxy` was defined. No proxy mocking should be required for that (?)

Reply:
Ok, I added a unit test.  Even after implementing the test, it still seems it's mainly testing `QNetworkProxy`, but it was easy enough to add, so it should be good.